include/alpaka/core/OmpSchedule.hpp

```diff
@@ -0,0 +1,94 @@
+/* Copyright 2020 Sergei Bastrakov
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/Common.hpp>
+
+#ifdef _OPENMP
+#    include <omp.h>
+#endif
+
+#include <cstdint>
+
+
+namespace alpaka
+{
+    namespace omp
+    {
+        //#############################################################################
+        //! Representation of OpenMP schedule information: kind and chunk size.
+        //!
+        //! This class can be used regardless of whether OpenMP is enabled
+        struct Schedule
+        {
+            //#############################################################################
+            //! Mirror of the OpenMP enum omp_sched_t with the corresponding integer values
+            enum Kind
+            {
+                Static = 1,
+                Dynamic = 2,
+                Guided = 3,
+                Auto = 4
+            };
+
+            //! Integer representation of schedule kind.
+            //!
+            //! Not stored as Kind to reduce casts when converting to and from omp_sched_t
+            uint32_t kind;
+
+            //! Chink size.
+            //!
+            //! Same as in OpenMP, value 0 corresponds to default chunk size
+            int chunkSize;
+
+            constexpr Schedule(Kind kind = Guided, int chunkSize = 0)
```

Comment:
Why do we need a ctor? `Schedule` is just an aggregate.

Reply:
Maybe for the default parameters?

Reply:
Well, then just directly initialize the members:
```c++
struct Schedule {
    Kind kind = Kind::Guided;
    int chunkSize = 0;
};
```

Reply:
I started with your suggestion, But it felt like too many casts when this struct is used, and so I went for storng internally as int and converting in a constructor. This is the only purpose of it.

Reply:
Will the cast really have a noticeable performance impact? I doubt that users will do high-frequency scheduler switching. Staying with `enum` will eliminate this issue anyway as soon as we transition to C++17 because then we can just write `enum (class) Kind : std::uint32_t`.

Reply:
No, performance does not matter here. Just a (subjective) matter of how clean the implementation looks